modules/social_features/social_private_message/social_private_message.install

```diff
@@ -359,21 +359,115 @@ function social_private_message_update_11005(): void {
+  $configs = [
+    'core.entity_form_display.message.private_message_notification.default',
+    'core.entity_view_display.message.private_message_notification.default',
+    'core.entity_view_display.message.private_message_notification.mail_body',
+    'core.entity_view_display.message.private_message_notification.mail_subject',
+    'field.field.message.private_message_notification.field_message_pm_thread',
+    'field.field.message.private_message_notification.field_message_private_message',
+    'field.storage.message.field_message_pm_thread',
+    'field.storage.message.field_message_private_message',
+    'message.template.private_message_notification',
+  ];
+
+  $config_factory = \Drupal::configFactory();
+
+  // Some users could edit the configuration, so we need to save it temporarily
+  // and update after private_message_notify will be enabled.
+  // @see social_private_message_update_11010
+  foreach ($configs as $name) {
+    $config = $config_factory->getEditable($name);
+    if ($config->isNew()) {
+      continue;
+    }
+
+    // Set config data to a new temporary config.
+    $config_tmp = $config_factory->getEditable("{$name}.tmp");
+    $config_tmp->setData($config->getRawData())->save();
+
+    // Delete config.
     $config->delete();
   }
 }
 
+/**
+ * Enable notifications submodule.
+ */
+function social_private_message_update_11009(): void {
+  $module = 'private_message_notify';
+  if (!\Drupal::moduleHandler()->moduleExists($module)) {
+    /** @var \Drupal\Core\Extension\ModuleInstallerInterface $installer */
+    $installer = \Drupal::service('module_installer');
+    $installer->install([$module]);
+  }
+}
+
+/**
+ * Update the configs that were migrated to private_message_notify.
+ */
+function social_private_message_update_11010(): void {
+  $configs = [
+    'core.entity_form_display.message.private_message_notification.default',
+    'core.entity_view_display.message.private_message_notification.default',
+    'core.entity_view_display.message.private_message_notification.mail_body',
+    'core.entity_view_display.message.private_message_notification.mail_subject',
+    'field.field.message.private_message_notification.field_message_pm_thread',
+    'field.field.message.private_message_notification.field_message_private_message',
+    'field.storage.message.field_message_pm_thread',
+    'field.storage.message.field_message_private_message',
+    'message.template.private_message_notification',
+  ];
+
+  $config_factory = \Drupal::configFactory();
+
+  foreach ($configs as $name) {
+    // Load temporary config.
+    $config_tmp = $config_factory->getEditable("{$name}.tmp");
+
+    // If there is not exist a temporary config, do nothing.
+    if ($config_tmp->isNew()) {
+      continue;
+    }
+
+    // Load temporary config and update original one.
+    $config = $config_factory->getEditable($name);
+    $config->setData($config_tmp->getRawData())->save();
```

Comment:
What if the config doesn't match the new config and it "breaks" on saving, how do we treat that potential exception?

For example, the temporary config had the following:

```
langcode: en
status: true
dependencies: {  }
template: private_message_notification
label: 'Private Message Notification'
description: 'The notification from a private message'
text:
  -
    value: "Private message at [site:name]\r\n"
    format: basic_html
  -
    value: "[user:display-name],\r\n\r\nYou have received a private message at [site:name] from [private_message:author-name]\r\n\r\nThe message is as follows:\r\n\r\n[private_message:message]\r\n\r\nYou can view the entire thread and reply to this message at:\r\n\r\n[private_message_thread:url]\r\n\r\n\r\nThank you,\r\n\r\n-- [site:name]\r\n"
    format: basic_html
settings:
  'token options':
    clear: false
    'token replace': true
  purge_override: false
```

In the meantime, the structure of the `message.template.*` changed. E.g. the entire Config structure / entity / schema was changed in the meantime (you don't know when this update happens)

You can't just save the raw data and expect it to work without any exception.

Reply:
I have added `try catch` to treat potential exceptions.